unstructured_ingest/v2/interfaces/processor.py

```diff
@@ -1,24 +1,24 @@
 import os
 from asyncio import Semaphore
-from dataclasses import dataclass, field
 from pathlib import Path
-from typing import Optional
+from typing import Any, Optional
 
-from unstructured_ingest.enhanced_dataclass import EnhancedDataClassJsonMixin
+from pydantic import BaseModel, ConfigDict, Field
 
 DEFAULT_WORK_DIR = str((Path.home() / ".cache" / "unstructured" / "ingest" / "pipeline").resolve())
 
 
-@dataclass
-class ProcessorConfig(EnhancedDataClassJsonMixin):
+class ProcessorConfig(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
what is the purpose of this? I saw it in extract_config but it doesn't seem to do anything.

Reply:
This allows the base config to contain content that isn't supported by pydantic, such as the Semaphore in this one.